src/lib/proof_carrying_data/proof_carrying_data.ml

```diff
@@ -6,7 +6,7 @@ open Core_kernel
 module Stable = struct
   module V1 = struct
     type ('a, 'p) t = ('a, 'p) Mina_wire_types.Proof_carrying_data.V1.t =
-      { data : 'a; proof : 'p }
+      { data : 'a [@key "statement"]; proof : 'p }
```

Comment:
I believe this change fully fixes the issue with the mina wire types modification.

The only possible troubles that I envision come from different name (`statement` vs `data`) in json and sexp serialization.

This change fully addresses it.

P.S. we don't have tests for sexp/json serialization that would capture it. I ran a manual test for json roundtrip using a precomputed block from itn-1 cluster (with siome zkapp txs, payments and completed works). This test is not obvious how to run in CI because of the file size, so it's for later. But the test results that I obtained locally confirm that `[@key "statement"]` is necessary and sufficient to make json serialization backwards-compatible.

Reply:
It's safe to make this change. Previously this data type wasn't provided a yojson instance, so nothing
will be broken.
The only use of it that requires yojson instances is for the precomputed blocks.

Reply:
P.S. binio serialization was fine even before the fixup, which is attested by our binio-shape-based unit test (which is extremely stringent).

Reply:
I didn't know field names are not serialized in binio

Reply:
It would be a waste to have them there, thanks god they are not serialized